internal/controllers/clusterextension_controller.go

```diff
@@ -580,7 +586,8 @@ func (r *ClusterExtensionReconciler) generateBundleDeploymentForUnpack(bundlePat
 			Source: rukpakv1alpha2.BundleSource{
 				Type: rukpakv1alpha2.SourceTypeImage,
 				Image: &rukpakv1alpha2.ImageSource{
-					Ref: bundlePath,
+					Ref:                   bundlePath,
+					InsecureSkipTLSVerify: true,
```

Comment:
Do we really want to hard code skipping this? Maybe this should come in from a flag as an option?

Reply:
Perhaps. Right now, it's necessary., unless we get CAs.

Reply:
What fails without this?

Reply:
Looks to have been used for debug and is not necessary.

Reply:
Unresolving as this needs attention.

Reply:
This should be a knob, and we should expose it an API, as it very much depends on where catalog is present, which in-turn depends on where v1 is running. 

If my understanding is correct - there is a draft PR which I have opened to discuss the change where we introduce this in a separate `ClusterExtensionSource`. 

cc: @tmshort @joelanford

Reply:
Is it required for rukpak bump? I was able to successfully run e2e in this PR without this change.

Reply:
It does fail e2es in CI

Reply:
That's weird. What is the difference between `make test-e2e` in CI and locally?

Reply:
But I think there may be a work around for this while setting up the remote registry... looking into it

Reply:
@varshaprasad96 makes sense. I figured out why I got a "pass" locally: because my test results got cached locally after a successful run with `InsecureSkipTLSVerify` set to `true`:

```
PASS
ok  	github.com/operator-framework/operator-controller/test/e2e	(cached)
```

Anyway - I feel a bit uneasy about changing the code to make tests work. Especially when it comes to TLS. You mentioned that there is a potentially a work around/fix for this. If we can fix that on test setup level - that would be great.

Reply:
Based on the discussion on slack, adding an annotation on CE to allow users to set `insecureSkipTLSVerify`. Will create a follow up, for an API design to include this in the spec.

Reply:
I didn't manage to find the discussion on slack you are referring to. Could you please share a link?

> Looking into the setup for spinning up the local registry, we do not configure any certificates so this seems expected.

For unrelated reason I was looking at the registry. It looks like we actually create certs for the registry with cert-manager, but they are self-signed and operator-controller knows nothing about their CA so it can't verify. We might want to look into #888 and see if we can make it more generic so that operator-controller is configurable with CAs.

Reply:
@m1kola  This is the thread: https://redhat-internal.slack.com/archives/C06KP34REFJ/p1717787814237379

tl; dr; Yes, the set-up does involve creating certs using cert-manager (I missed looking at it previously). It involves creating certs and having them in secrets (https://github.com/operator-framework/operator-controller/blob/5687a69155e3c6e279ea0c304c9d83e5c11369c1/test/tools/image-registry.sh#L82-L89) which is then specified in the registry pod. However, the operator-controller pod currently does not have a volume to mount those secrets and all the deployments in this case are on different namespaces making it difficult to share. On exploring various options to go about this, the fix for now was to introduce an API change allowing users to pass an option to specify tls configuration. This is a little complicated to do in this PR, as we would have to discuss how the API would look like if we add it to the spec- to make it compatible with future options (like direct bundle install). Hence we decided to keep this PR moving and to have a super minimalistic change to allow users skip tls verification through an annotation.

Reply:
Thanks for the link and context.

Just to clarify - when I mentioned #888 and looking into how we can make operator-controller be aware of CAs, etc - I did not mean that we need to do it in this PR.

It is more about the follow up work - as an option.

Reply:
We discussed adding a method to get CAs into operator-controller, and they are all _too big_ to be done here. It will require some additional thought, and it requires modifying the tests to do things in a proper order.
Specifically, we need to have any cert-manager secrets created _before_ operator controller, and right now, operator-controller is created before any of the e2e infrastructure, which includes the docker-registry and corresponding Secret.
So, we are adding an annotation to make this configurable on a per-ClusterExtension basis, and will subsequently update either the CE API, the operator-controller pod spec, or both.